CONTRIBUTING.md

```diff
@@ -79,6 +85,23 @@ an AST expression). `match_def_path()` in Clippy's `utils` module can also be us
 [if_chain]: https://docs.rs/if_chain/*/if_chain
 [nest-less]: https://github.com/rust-lang/rust-clippy/blob/557f6848bd5b7183f55c1e1522a326e9e1df6030/clippy_lints/src/bit_mask.rs#L124-L150
 
+### How to build Clippy
+
+Clippy relies on rustc internals, so you'll need to set up your Rustup toolchain.
+**Follow the instructions in the [Basics docs](doc/basics.md) to get set up,
+such as by running the `setup-toolchain.sh` script.**
+
+Once you've run `setup-toolchain.sh`, you can use `cargo build` and
+`cargo test` like any other Rust project!
+
+**Note:** You may get compiler errors when building Clippy, even if you
+didn't change anything. That's because, as mentioned earlier, Clippy
+depends on rustc internals. There is [discussion][nightly-build-issue]
+on ways to make sure that rustc does not break Clippy in the future
+so that Clippy always builds.
+
+[nightly-build-issue]: https://github.com/rust-lang/rust-clippy/issues/5561
+
```

Comment:
I **really** don't want this here. Keeping the docs up to date is already hard enough as it is. -> DRY principle.

Reply:
I think it might be helpful to move the note here to the basics docs, then, so that new contributors aren't confused by the compiler errors. Does that make sense to you?

Reply:
Not really. Normally these errors are fixed within a few hours and most of the contributors don't see them. This is just unlucky timing, since we're waiting for other Clippy changes to get merged in rustc, before fixing those. 

In addition to that, the basics docs should be self contained and contain all the information needed to build Clippy. The split of these information between multiple documents was the reason the basics docs exist in the first place.

Reply:
I just think it might be a bit confusing for people who are new clippy and don't understand tool breakages. Is there somewhere that you would feel comfortable putting it?

Reply:
If not, I can remove it, I just thought it might be helpful

Reply:
Oh I thought this was already mentioned in `basics.md`. You can put it here: https://github.com/rust-lang/rust-clippy/blob/master/doc/basics.md#setup

Reply:
Hmm, not under https://github.com/rust-lang/rust-clippy/blob/master/doc/basics.md#building-and-testing?

Reply:
Ah, did you suggest that since that part mentions how Clippy depends on internals?

Reply:
> Ah, did you suggest that since that part mentions how Clippy depends on internals?

Yes exactly. That way we can keep this note a little shorter, and it is higher up in the document ;)